### PR TITLE
Created Bank application

### PR DESCRIPTION
**Pull request by Lukk1515**

### CODE REVIEW

File: src/bank_app/bank_app.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+class BankAccount:
+    def __init__(self, balance, account_number, account_holder):
```
Comment (dominikjagoda) [inline src/bank_app/bank_app.py:2]: Please add typing to `__init__()` arguments

Comment (Lukk1515) [reply src/bank_app/bank_app.py]: [Inproved typing in bank_app.py](https://github.com/Lukk1515/bank_app/pull/1/commits/223fb43cf77dd5526374c97b33272e2170c2f049)

File: src/bank_app/bank_app.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        self.__balance = balance
+        self.__account_number = account_number
+        self.__account_holder = account_holder
+        self.__transaction_history = []
+        self.__overdraft_limit = 0
+
+    def get_balance(self):
+        return self.__balance
+
+    def add_balance(self, amount):
```
Comment (dominikjagoda) [inline src/bank_app/bank_app.py:12]: Missing typing in function arguments

File: src/bank_app/bank_app.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        self.__balance += amount
+        self.__transaction_history.append(f"Deposited {amount}")
+        print(f"Deposited {amount}. Current balance: {self.__balance}")
+
+    def card_payment(self, price):
```
Comment (dominikjagoda) [inline src/bank_app/bank_app.py:17]: Missing typing in function arguments

File: src/bank_app/bank_app.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            raise ValueError("Transaction rejected. No funds in account.")
+
+        self.__balance -= price
+        self.__transaction_history.append(f"Withdraw {price}")
+        print(f"Withdraw {price}. Current balance: {self.__balance}")
+
+    def get_transaction_history(self):
+        return "\n".join(self.__transaction_history)
+
+    def transfer(self, amount, other_account):
```
Comment (dominikjagoda) [inline src/bank_app/bank_app.py:31]: Missing typing in function arguments

File: src/bank_app/bank_app.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        print(
+            f"Transferred {amount} to account {other_account.__account_number}. Current balance: {self.__balance}"
+        )
+
+    def print_statement(self):
+        print(
+            f"Completed transactions: {self.__transaction_history}. Current balance: {self.__balance}"
+        )
+
+    def apply_interest(self, interest_rate):
```
Comment (dominikjagoda) [inline src/bank_app/bank_app.py:54]: Missing typing in function arguments

File: src/bank_app/bank_app.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        interest = self.__balance * interest_rate
+        self.__balance += interest
+        self.__transaction_history.append(
+            f"The accrued interest is {interest}. Current balance: {self.__balance}"
+        )
+        print(f"The account balance after adding interest is {self.__balance}")
+
+    def change_account_holder(self, new_holder):
```
Comment (dominikjagoda) [inline src/bank_app/bank_app.py:62]: Missing typing in function arguments

File: src/bank_app/bank_app.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        self.__account_holder = new_holder
+        self.__transaction_history.append(f"New holder the account is {new_holder}")
+        print(f"New holder the account is {new_holder}")
+
+    def set_overdraft_limit(self, limit):
```
Comment (dominikjagoda) [inline src/bank_app/bank_app.py:67]: Missing typing in function arguments

File: src/bank_app/bank_app.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
+class BankAccount:
+    def __init__(self, balance: float, account_number: str, account_holder: str):
```
Comment (dominikjagoda) [inline src/bank_app/bank_app.py:2]: Please add docstirngs to class method

File: src/bank_app/bank_app.py [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+        self.__balance = balance
+        self.__account_number = account_number
+        self.__account_holder = account_holder
+        self.__transaction_history = []
+        self.__overdraft_limit = 0
+
+    def get_balance(self):
```
Comment (dominikjagoda) [inline src/bank_app/bank_app.py:9]: Please add docstirngs to class method

File: src/bank_app/bank_app.py (hunk 2 above)
Comment (dominikjagoda) [inline src/bank_app/bank_app.py:12]: Please add docstirngs to class method

File: src/bank_app/bank_app.py (hunk 3 above)
Comment (dominikjagoda) [inline src/bank_app/bank_app.py:17]: Please add docstirngs to class method

File: src/bank_app/bank_app.py [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+        transaction = self.__balance - price
+
+        if transaction < 0:
+            raise ValueError("Transaction rejected. No funds in account.")
+
+        self.__balance -= price
+        self.__transaction_history.append(f"Withdraw {price}")
+        print(f"Withdraw {price}. Current balance: {self.__balance}")
+
+    def get_transaction_history(self):
```
Comment (dominikjagoda) [inline src/bank_app/bank_app.py:28]: Please add docstirngs to class method

File: src/bank_app/bank_app.py [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+            f"Transferred {amount} to account {other_account.__account_number}"
+        )
+        other_account.__transaction_history.append(
+            f"Received {amount} from account {self.__account_number}"
+        )
+        print(
+            f"Transferred {amount} to account {other_account.__account_number}. Current balance: {self.__balance}"
+        )
+
+    def print_statement(self):
```
Comment (dominikjagoda) [inline src/bank_app/bank_app.py:49]: Please add docstirngs to class method

File: src/bank_app/bank_app.py (hunk 5 above)
Comment (dominikjagoda) [inline src/bank_app/bank_app.py:54]: Please add docstirngs to class method

File: src/bank_app/bank_app.py (hunk 6 above)
Comment (dominikjagoda) [inline src/bank_app/bank_app.py:62]: Please add docstirngs to class method

File: src/bank_app/bank_app.py (hunk 7 above)
Comment (dominikjagoda) [inline src/bank_app/bank_app.py:67]: Please add docstirngs to class method

**Review by dominikjagoda — CHANGES_REQUESTED** [review]
Please add docstrings and typing in `bank_app.py`

File: src/bank_app/bank_app.py (hunk 7 above)
Comment (Lukk1515) [reply src/bank_app/bank_app.py:114]: [Added docstrins in bank_app.py](https://github.com/Lukk1515/bank_app/pull/1/commits/2e50a2b4fe468c2774318e44a563e8d8d809876a)

File: src/bank_app/bank_app.py (hunk 6 above)
Comment (Lukk1515) [reply src/bank_app/bank_app.py:104]: [Added docstrins in bank_app.py](https://github.com/Lukk1515/bank_app/pull/1/commits/2e50a2b4fe468c2774318e44a563e8d8d809876a)

File: src/bank_app/bank_app.py (hunk 5 above)
Comment (Lukk1515) [reply src/bank_app/bank_app.py:91]: [Added docstrins in bank_app.py](https://github.com/Lukk1515/bank_app/pull/1/commits/2e50a2b4fe468c2774318e44a563e8d8d809876a)

File: src/bank_app/bank_app.py (hunk 11 above)
Comment (Lukk1515) [reply src/bank_app/bank_app.py:86]: [Added docstrins in bank_app.py](https://github.com/Lukk1515/bank_app/pull/1/commits/2e50a2b4fe468c2774318e44a563e8d8d809876a)

File: src/bank_app/bank_app.py (hunk 10 above)
Comment (Lukk1515) [reply src/bank_app/bank_app.py:52]: [Added docstrins in bank_app.py](https://github.com/Lukk1515/bank_app/pull/1/commits/2e50a2b4fe468c2774318e44a563e8d8d809876a)

File: src/bank_app/bank_app.py (hunk 3 above)
Comment (Lukk1515) [reply src/bank_app/bank_app.py:34]: [Added docstrins in bank_app.py](https://github.com/Lukk1515/bank_app/pull/1/commits/2e50a2b4fe468c2774318e44a563e8d8d809876a)

File: src/bank_app/bank_app.py (hunk 2 above)
Comment (Lukk1515) [reply src/bank_app/bank_app.py:24]: [Added docstrins in bank_app.py](https://github.com/Lukk1515/bank_app/pull/1/commits/2e50a2b4fe468c2774318e44a563e8d8d809876a)

File: src/bank_app/bank_app.py (hunk 9 above)
Comment (Lukk1515) [reply src/bank_app/bank_app.py:16]: [Added docstrins in bank_app.py](https://github.com/Lukk1515/bank_app/pull/1/commits/2e50a2b4fe468c2774318e44a563e8d8d809876a)

File: src/bank_app/bank_app.py (hunk 8 above)
Comment (Lukk1515) [reply src/bank_app/bank_app.py:2]: [Added docstrins in bank_app.py](https://github.com/Lukk1515/bank_app/pull/1/commits/2e50a2b4fe468c2774318e44a563e8d8d809876a)

File: src/bank_app/bank_app.py (hunk 2 above)
Comment (Lukk1515) [reply src/bank_app/bank_app.py]: [Added typing arguments in functions](https://github.com/Lukk1515/bank_app/pull/1/commits/6d8e7b4b6084737013808da1fd4eef2147da6fad)

File: src/bank_app/bank_app.py (hunk 3 above)
Comment (Lukk1515) [reply src/bank_app/bank_app.py]: [Added typing arguments in functions](https://github.com/Lukk1515/bank_app/pull/1/commits/6d8e7b4b6084737013808da1fd4eef2147da6fad)

File: src/bank_app/bank_app.py (hunk 4 above)
Comment (Lukk1515) [reply src/bank_app/bank_app.py]: [Added typing arguments in functions](https://github.com/Lukk1515/bank_app/pull/1/commits/6d8e7b4b6084737013808da1fd4eef2147da6fad)

File: src/bank_app/bank_app.py (hunk 5 above)
Comment (Lukk1515) [reply src/bank_app/bank_app.py]: [Added typing arguments in functions](https://github.com/Lukk1515/bank_app/pull/1/commits/6d8e7b4b6084737013808da1fd4eef2147da6fad)

File: src/bank_app/bank_app.py (hunk 6 above)
Comment (Lukk1515) [reply src/bank_app/bank_app.py]: [Added typing arguments in functions](https://github.com/Lukk1515/bank_app/pull/1/commits/6d8e7b4b6084737013808da1fd4eef2147da6fad)

File: src/bank_app/bank_app.py (hunk 7 above)
Comment (Lukk1515) [reply src/bank_app/bank_app.py]: [Added typing arguments in functions](https://github.com/Lukk1515/bank_app/pull/1/commits/6d8e7b4b6084737013808da1fd4eef2147da6fad)

**Pull request merged**